### DISCUSSION TITLE
Regex parsing for autodiscoverFilter does not work in all cases

### DISCUSSION THREAD

**Discussion by gaeljw** [renovatebot/renovate#23977] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.49.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm setting up a `autodiscoverFilter` with a regex and I can't make the regex work.

The regex I'd like to use: `myGroup\/((?!myProject).)*`. The purpose of this regex is to autodiscover all projects of the group but a single one.

I tried several variations in the `autodiscoverFilter` but all of them end up with the following error:
```json
{
           "name": "renovate",
           "level": 60,
           "logContext": "D6kRj68psoQN6gSCA3fqo",
           "err": {
             "message": "Failed to parse regex pattern \"/myGroup/((?!myProject).)*/\"",
             "stack": "Error: Failed to parse regex pattern \"/myGroup/((?!myProject).)*/\"\n    at applyFilters (/opt/containerbase/tools/renovate/36.49.0/node_modules/renovate/lib/workers/global/autodiscover.ts:105:15)\n    at autodiscoverRepositories (/opt/containerbase/tools/renovate/36.49.0/node_modules/renovate/lib/workers/global/autodiscover.ts:51:18)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at start (/opt/containerbase/tools/renovate/36.49.0/node_modules/renovate/lib/workers/global/index.ts:151:14)\n    at /opt/containerbase/tools/renovate/36.49.0/node_modules/renovate/lib/renovate.ts:18:22"
           },
           "msg": "Fatal error: Failed to parse regex pattern \"/myGroup/((?!myProject).)*/\""
         }
```

What I tried:
- `autodiscoverFilter: ["/myGroup\/((?!myProject).)*/"]`
- `autodiscoverFilter: ["/myGroup/((?!myProject).)*/"]`
- `autodiscoverFilter: ["/myGroup\\/((?!myProject).)*/"]`

Is this something not supported by Renovate or maybe I need some kind of escape in my regex?

### Relevant debug logs

See above

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment, marked as answer]
we're using re2 as regex engine, so you can't use look ahead or behind
